Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -0,0 +1,175 @@
+
+import XCTest
+
+import NIOCore
+import NIOEmbedded
+import NIOHPACK
+import NIOHTTP1
+@_spi(AsyncChannel) import NIOHTTP2
+import NIOTLS
+
+final class ConfiguringPipelineAsyncMultiplexerTests: XCTestCase {
+    var clientChannel: NIOAsyncTestingChannel!
+    var serverChannel: NIOAsyncTestingChannel!
+
+    override func setUp() {
+        self.clientChannel = NIOAsyncTestingChannel()
+        XCTAssertNoThrow(try self.clientChannel.connect(to: .init(unixDomainSocketPath: "ignored")).wait())
+        self.serverChannel = NIOAsyncTestingChannel()
+        XCTAssertNoThrow(try self.serverChannel.connect(to: .init(unixDomainSocketPath: "ignored")).wait())
+    }
+
+    override func tearDown() {
+        self.clientChannel = nil
+        self.serverChannel = nil
+    }
+
+
+    static let requestFramePayload = HTTP2Frame.FramePayload.headers(.init(headers: HPACKHeaders([(":method", "GET"), (":authority", "localhost"), (":scheme", "https"), (":path", "/")]), endStream: true))
+    static let responseFramePayload = HTTP2Frame.FramePayload.headers(.init(headers: HPACKHeaders([(":status", "200")]), endStream: true))
+
+    class OKResponder: ChannelInboundHandler {
+        typealias InboundIn = HTTP2Frame.FramePayload
+        func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            let frame = self.unwrapInboundIn(data)
+            switch frame {
+            case .headers:
+                break
+            default:
+                fatalError("unexpected frame type: \(frame)")
+            }
+
+            context.channel.writeAndFlush(responseFramePayload, promise: nil)
+
+            context.fireChannelRead(data)
+        }
+    }
+
+    class SimpleRequest: ChannelInboundHandler, ChannelOutboundHandler {
+        typealias InboundIn = HTTP2Frame.FramePayload
+        typealias OutboundIn = HTTP2Frame.FramePayload
+
+        func writeRequest(context: ChannelHandlerContext) {
+            context.channel.writeAndFlush(requestFramePayload, promise: nil)
+        }
+
+        func channelActive(context: ChannelHandlerContext) {
+            self.writeRequest(context: context)
+            context.fireChannelActive()
+        }
+
+        func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            let frame = self.unwrapInboundIn(data)
+            switch frame {
+            case .headers:
+                break
+            default:
+                fatalError("unexpected frame type: \(frame)")
+            }
+            context.fireChannelRead(data)
+        }
+    }
+
+    func testBasicPipelineCommunicates() async throws {
+        let requestCount = 100
+
+
+        let (serverRecorderStream, serverRecorderContinuation) = AsyncStream<InboundRecorder<HTTP2Frame.FramePayload>>.makeStream()
+        var serverRecorderIterator = serverRecorderStream.makeAsyncIterator()
+
+        let clientMultiplexer = try await assertNoThrowWithValue(
```

Comment:
FWIW: `XCTAssertNoThrow` was valuable because Xcode provided pretty bad errors when something was thrown; the errors are much better now so I tend to think `XCTAssertNoThrow` (and by extension `assertNoThrowWithValue`) aren't really necessary. There's no need to remove it but I won't ask you to use it if you don't!